### PR TITLE
fix(closure): prevent closure renaming of testability interface

### PR DESCRIPTION
**Pull request by alexeagle**
Needed to enable closure compiler ADVANCED_OPTIMIZATIONS

### CODE REVIEW

**Comment by juliemr** [comment]
discussed offline - will instead move `PublicTestability` to core and export it there.

**Comment by alexeagle** [comment]
FYI @IgorMinar re. the public API shape change

File: modules/@angular/core/src/testability/testability.ts [hunk 1]
```diff
@@ -11,7 +11,18 @@ import {Map, MapWrapper} from '../facade/collection';
...
+/**
+ * Testability API.
+ * `declare` keyword causes tsickle to generate externs, so these methods are
+ * not renamed by Closure Compiler.
+ * @experimental
+ */
+export declare interface PublicTestability {
+  isStable(): boolean;
+  whenStable(callback: Function): void;
+  findBindings(using: any, provider: string, exactMatch: boolean): any[];
```
Comment (juliemr) [inline modules/@angular/core/src/testability/testability.ts]: Let's go ahead and kill findBindings, it should have been deprecated.

**Comment by alexeagle** [comment]
per @IgorMinar I removed the new interface from the re-exports in `core/index`

**Pull request merged**